feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIcalls.java

```diff
@@ -0,0 +1,158 @@
+            }
+            con.getInputStream();
+    }
+
+    public void update_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+            String name = getname(patientFR);
+            String uuid = getuuid(patientFR);
+
+            String id = this.get_list_customers(uuid, auth);
+            HttpURLConnection con = getConnection("customers/" + id, "PUT", auth);
+
+            JSONObject parameters = new JSONObject();
+            parameters.put("name", name);
+            String jsonInputString = parameters.toString();
+
+            try (OutputStream os = con.getOutputStream()) {
+                byte[] inputString = jsonInputString.getBytes(UTF_8);
+                os.write(inputString, 0, inputString.length);
+            }
+            con.getInputStream();
+    }
+
+    public String get_list_customers(String uuid, String auth) throws URISyntaxException, JSONException, IOException {
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpUriRequest list_customers = RequestBuilder.get()
+                .setUri(new URI(properties.getUrl()+ "/api/v1/customers"))
+                .addParameter("contact_name", uuid)
+                .build();
+
+        list_customers.addHeader("Authorization", "Bearer " + auth);
+        list_customers.addHeader("Accept", "*/*");
+        list_customers.addHeader("Content-Type", "*/*");
+        list_customers.addHeader("company", "1");
+
+        CloseableHttpResponse response = httpclient.execute(list_customers);
+        JSONObject myObject = new JSONObject(EntityUtils.toString(response.getEntity()));
+
+        if (myObject.getJSONArray("data").length() == 0) {
+            return "Customer not found";
+
+        } else {
+            return String.valueOf(myObject.getJSONArray("data").getJSONObject(0).getInt("id"));
+        }
+    }
+
+    public void create_update(OpenMRSPatientFullRepresentation patientFR) throws Exception {
+        if (login_verification(auth).equals("Authenticated")) {
+            String uuid = getuuid(patientFR);
+            if (get_list_customers(uuid, auth).equals("Customer not found")) {
+                create_customer(patientFR, auth);
+            } else {
+                update_customer(patientFR, auth);
+            }
+        }
+        else{
+            logger.error("Not Authenticated");
+        }
+    }
+
+    public String getuuid(OpenMRSPatientFullRepresentation patientFR) {
+        Optional<OpenMRSPatientIdentifier> uuid1 = patientFR.getIdentifiers().stream()
+                .filter(OpenMRSPatientIdentifier::isPreferred)
+                .findFirst();
+
+        return uuid1.isPresent() ? uuid1.get().getIdentifier() : EMPTY_STRING;
+    }
+
+    public String getname(OpenMRSPatientFullRepresentation patientFR) {
+        OpenMRSPersonName preferredName = patientFR.getPerson().getPreferredName();
+        return preferredName.getGivenName()
```

Comment:
This logic should be in OpenMRSPersonName class with a method called: getPreferredFullName( ).